wincode-derive/src/schema_read.rs

```diff
@@ -225,27 +230,232 @@ fn impl_struct_extensions(args: &SchemaArgs) -> Result<TokenStream> {
+                    // SAFETY: We are dropping an initialized field.
+                    unsafe {
+                        ptr::drop_in_place(&raw mut (*dst_ptr).#field_ident);
+                    }
+                }
+            }
+        });
+        quote! {
+            impl #builder_impl_generics Drop for #builder_ident #builder_ty_generics #builder_where_clause {
+                fn drop(&mut self) {
+                    let dst_ptr = self.inner.as_mut_ptr();
+                    #(#drops)*
+                }
+            }
+        }
+    };
+
+    let builder_impl = quote! {
+        impl #builder_impl_generics #builder_ident #builder_ty_generics #builder_where_clause {
+            #vis const fn from_maybe_uninit_mut(inner: &'_wincode_inner mut MaybeUninit<#builder_dst>) -> Self {
+                Self {
+                    inner,
+                    init_set: 0,
+                }
+            }
+
+            /// Assume the builder is fully initialized, and return a mutable reference to the inner `MaybeUninit` struct.
+            ///
+            /// The builder will be forgotten, so the drop logic will not longer run.
+            ///
+            /// # Safety
+            ///
+            /// Calling this when the content is not yet fully initialized causes undefined behavior: it is up to the caller
+            /// to guarantee that the `MaybeUninit<T>` really is in an initialized state.
+            #[inline]
+            #vis const unsafe fn assume_init_mut(mut self) -> &'_wincode_inner mut #builder_dst {
+                // SAFETY: reference lives beyond the scope of the builder, and builder is forgotten.
+                let inner = unsafe { ptr::read(&mut self.inner) };
+                mem::forget(self);
+                // SAFETY: Caller asserts the `MaybeUninit<T>` is in an initialized state.
+                unsafe {
+                    inner.assume_init_mut()
+                }
+            }
+
+            /// Forget the builder, disabling the drop logic.
+            ///
+            /// # Safety
+            ///
+            /// Assuming the source `MaybeUninit<T>` when the content is not yet fully initialized causes undefined behavior.
+            /// It is up to the caller to guarantee that the `MaybeUninit<T>` really is in an initialized state.
+            #[inline]
+            #vis const unsafe fn assume_init_forget(self) {
+                mem::forget(self);
+            }
+        }
+    };
+
+    // Generate the helper methods for the builder.
+    let builder_helpers = fields.iter().enumerate().map(|(i, field)| {
+        let target = field.target_resolved();
+        let target_reader_bound = target.with_lifetime("de");
+        let ident = field.struct_member_ident(i);
+        let ident_string = field.struct_member_ident_to_string(i);
+        let uninit_mut_ident = format_ident!("uninit_{ident_string}_mut");
+        let read_field_ident = format_ident!("read_{ident_string}");
```

Comment:
btw, is there some mechanism that prevents those generated functions form triggering "unused" lint errors? I know this is an annoying issue with `modular_bitfield` crate code-gen.